src/components/ToggleButtonHeader.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useGlobalState } from '../CustomHooks/useGlobalState';
+import { useDispatch } from '../CustomHooks/useDispatch';
+import { Todo } from '../types/Todo';
+import { patchTodoFromStorage } from '../api/todos';
+import classNames from 'classnames';
+
+export const ToggleButtonHeader: React.FC = () => {
+  const { todos } = useGlobalState();
+  const dispatch = useDispatch();
+
+  const isAllTodosCompleted = todos.every(todo => todo.completed === true);
+
+  const promiseToggleCompleted = ({ id, completed, title }: Todo) => {
+    return new Promise<void>(resolve => {
```

Comment:
Why Promise? you can do it in a simple function

Reply:
because of the localStorage